api-inference-community/docker_images/keras/app/pipelines/image_classification.py

```diff
@@ -1,35 +1,51 @@
 import json
-from typing import TYPE_CHECKING, Any, Dict, List
+from typing import Any, Dict, List
 
 import tensorflow as tf
 from app.pipelines import Pipeline
-from huggingface_hub import cached_download, hf_hub_url
+from huggingface_hub import cached_download, from_pretrained_keras, hf_hub_url
+from PIL import Image
 
 
-if TYPE_CHECKING:
-    from PIL import Image
+# PIL Interpolation Methods - More info can be found in the link below
+# https://pillow.readthedocs.io/en/stable/handbook/concepts.html#filters-comparison-table
+_PIL_INTERPOLATION_METHODS = {
+    "nearest": Image.NEAREST,
+    "bilinear": Image.BILINEAR,
+    "bicubic": Image.BICUBIC,
+    "hamming": Image.HAMMING,
+    "box": Image.BOX,
+    "lanczos": Image.LANCZOS,
+}
 
-MODEL_FILENAME = "tf_model.h5"
+MODEL_FILENAME = "saved_model.pb"
 CONFIG_FILENAME = "config.json"
 
 
 class ImageClassificationPipeline(Pipeline):
     def __init__(self, model_id: str):
-        # Warning: this is a short term solution to get something working :)
-        # This loading has strong assumptions and should be replaced with a better
-        # wrapper for this. This assumes the full model is saved, not just
-        # the weights.
-        model_file = cached_download(hf_hub_url(model_id, filename=MODEL_FILENAME))
-        self.model = tf.keras.models.load_model(model_file)
+
+        # Reload Keras SavedModel
+        self.model = from_pretrained_keras(model_id)
 
         # Handle binary classification with a single output unit.
         self.single_output_unit = self.model.output_shape[1] == 1
+        self.num_labels = 2 if self.single_output_unit else self.model.output_shape[1]
 
         # Config is required to know the mapping to label.
         config_file = cached_download(hf_hub_url(model_id, filename=CONFIG_FILENAME))
         with open(config_file) as config:
             config = json.load(config)
-        self.id2label = config["id2label"]
+
+        self.labels = config.get("labels", None)
```

Comment:
Any reason for this change? I think `id2label` is the usual approach in `transformers` and would prefer to keep consistent unless there's a strong reason.